### PR TITLE
return error from UpdateFromVolume in case of both attempt failed

### PR DESCRIPTION
**Pull request by giggsoff**
We [hit](https://github.com/lf-edge/eve/runs/4237527759?check_suite_focus=true) `OCI runtime create failed: args must not be empty: unknown` error and it indicates that we have empty spec's Process.Args.
In this PR I fix returning of error from UpdateFromVolume in case of both attempts of getting information failed (os.Open and getSavedImageInfo return errors). It may help to determine the problem.

Signed-off-by: Petr Fedchenkov <giggsoff@gmail.com>

### CODE REVIEW

File: pkg/pillar/containerd/oci.go [hunk 1]
```diff
@@ -315,20 +315,24 @@ func (s *ociSpec) UpdateFromDomain(dom *types.DomainConfig) {
 // OCI runtime spec takes precedence). In addition to that each volume is
 // expected to have rootfs subfolder with a full rootfs filesystem
 func (s *ociSpec) UpdateFromVolume(volume string) error {
+	//if not based on an OCI image, nothing to update here
+	if volume == "" {
+		return nil
+	}
```
Comment (eriknordmark) [inline pkg/pillar/containerd/oci.go:321]: Doesn't the caller already check this??

**Review by eriknordmark — APPROVED** [review]
Let it test

File: pkg/pillar/containerd/oci.go (hunk 1 above)
Comment (giggsoff) [reply pkg/pillar/containerd/oci.go:321]: No, and I hit that problem during my experiments with VMs

**Review by rvs — COMMENTED** [review]
LGTM

**Pull request merged**